docs/source/guides/data.rst

```diff
@@ -0,0 +1,355 @@
+
+.. testsetup:: *
+
+    from pytorch_lightning.core.Lightning import LightningModule
+    from torch.utils.data import IterableDataSet
+    from pytorch_lightning.trainer.trainer import Trainer
+
+.. _data:
+
+#############
+Managing Data
+#############
+
+Continue reading to learn about:
+
+* `<Data Containers in Lightning_>`_
+
+* `Iterate over multiple datasets <Multiple DataSets_>`_
+
+* `Handle sequential data <Sequential Data_>`_
+
+****************************
+Data Containers in Lightning
+****************************
+
+There are a few different Data containers used in Lightning:
+
+.. list-table:: Data objects
+   :widths: 20 80
+   :header-rows: 1
+
+   * - Object
+     - Definition
+   * - :class:`~torch.utils.data.Dataset`
+     - The PyTorch :class:`~torch.utils.data.Dataset` represents a map from keys to data samples.
+   * - :class:`~torch.utils.data.DataLoader`
+     - The PyTorch :class:`~torch.utils.data.DataLoader` represents a Python iterable over a DataSet.
+   * - :class:`~pytorch_lightning.core.datamodule.LightningDataModule`
+     - A :class:`~pytorch_lightning.core.datamodule.LightningDataModule` is simply a collection of a training DataLoader, validation DataLoader(s) and test DataLoader(s), along with the matching transforms and data processing/downloads steps required.
+
+Read :ref:`this <datamodules>` for more details on LightningDataModules.
+
+
+.. _multiple-training-dataloaders:
+
+*****************
+Multiple DataSets
+*****************
+
+There are a few ways to pass multiple DataSets to Lightning:
```

Comment:
The LightningModule was only mentioned once, and the first section here jumps right into multiple dataloaders. 
Wouldn't it make sense to introduce the DataModule first and give a motivation for using it?
After all, this document is under guides and indexed in the menu under "best practices".